CMakeLists.txt

```diff
@@ -31,6 +20,29 @@ option(USE_UBSAN "Enable UndefinedBehaviorSanitizer checks" OFF)
 option(USE_TSAN "Enable ThreadSanitizer checks" OFF)
 option(USE_MSAN "Enable MemorySanitizer checks" OFF)
 
+set(CXX_REQUIRING_OPTIONS
+    "UMF_BUILD_TESTS"
+    "UMF_BUILD_LIBUMF_POOL_DISJOINT")
+foreach(option_name ${CXX_REQUIRING_OPTIONS})
+    if(${option_name})
+        project(unified-memory-framework VERSION 0.1.0)
+        set(CMAKE_CXX_STANDARD 17)
+        set(CMAKE_CXX_STANDARD_REQUIRED YES)
+        break()
+    endif()
+endforeach()
+
+if(NOT CMAKE_CXX_STANDARD_REQUIRED)
+    project(unified-memory-framework VERSION 0.1.0 LANGUAGES C)
```

Comment:
Instead of keeping two lines with hardcoded version "0.1.0" it can be done via `enable_language()` and then only enable C, I think that it should be also enabled for C++ variant of the execution path.
Source: https://cmake.org/cmake/help/latest/command/enable_language.html

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e73bfa4..6f4fb4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

 cmake_minimum_required(VERSION 3.14.0 FATAL_ERROR)
+project(unified-memory-framework VERSION 0.1.0 LANGUAGES C)

 # Build Options
 option(UMF_BUILD_SHARED_LIBRARY "Build UMF as shared library" OFF)
@@ -25,17 +26,13 @@ set(CXX_REQUIRING_OPTIONS
     "UMF_BUILD_LIBUMF_POOL_DISJOINT")
 foreach(option_name ${CXX_REQUIRING_OPTIONS})
     if(${option_name})
-        project(unified-memory-framework VERSION 0.1.0)
+        enable_language(CXX)
         set(CMAKE_CXX_STANDARD 17)
         set(CMAKE_CXX_STANDARD_REQUIRED YES)
         break()
     endif()
 endforeach()

-if(NOT CMAKE_CXX_STANDARD_REQUIRED)
-    project(unified-memory-framework VERSION 0.1.0 LANGUAGES C)
-endif()
-
 include(CTest)
 include(CMakePackageConfigHelpers)
 include(GNUInstallDirs)
```

Reply:
Thanks, done.